### PR TITLE
DASH-190 DASH-196 Add Item Descriptors, Allow domain renewal and add bundles add to cart

### PR DESCRIPTION
**Pull request by vmkarpenko**

### CODE REVIEW

**Review by NikolasStelmakh — APPROVED** [review]

File: client/src/components/Cart/CartItem.tsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { CURRENCY_CODES } from '../../constants/common';
 import { CartItem as CartItemType, PaymentCurrency } from '../../types';
 
 import classes from './Cart.module.scss';
+import { getCartItemDescriptor } from '../../util/cart';
```
Comment (trukhilio) [inline client/src/components/Cart/CartItem.tsx:13]: Please, move above `types` import

File: client/src/components/ManagePageContainer/ManagePageComponents/ActionButtons.tsx [hunk 2]
```diff
@@ -18,14 +18,18 @@ import icon from '../../../assets/images/timelapse_white_24dp.svg'; // todo: rem
...
+  const handleAddBundles = () => {
+    onAddBundles(name);
+  };
 
   return (
     <div className={classes.actionButtonsContainer}>
-      <Link
-        to={`${ROUTES.FIO_ADDRESS_ADD_BUNDLES}?name=${name}`}
+      <div
+        onClick={handleAddBundles}
```
Comment (trukhilio) [inline client/src/components/ManagePageContainer/ManagePageComponents/ActionButtons.tsx:32]: Could we use the same style as for `onRenewDomain` or vice versa?

File: client/src/components/ManagePageContainer/ManagePageContainer.tsx [hunk 3]
```diff
@@ -15,8 +15,19 @@ import InfiniteScroll from '../InfiniteScroll/InfiniteScroll';
 import InfoBadge from '../Badges/InfoBadge/InfoBadge';
 
 import { BANNER_DATA, ITEMS_LIMIT, EXPIRED_DAYS, SUBTITLE } from './constants';
-import { useCheckIfDesktop } from '../../screenType';
+import { FIO_ADDRESS_DELIMITER } from '../../utils';
```
Comment (trukhilio) [inline client/src/components/ManagePageContainer/ManagePageContainer.tsx:18]: Move to `utils` import block

File: client/src/components/PurchaseNow/middleware.ts [hunk 4]
```diff
@@ -20,7 +24,9 @@ import {
   RegistrationRegistered,
   AnyObject,
   PurchaseTxStatus,
+  CartItemType,
 } from '../../types';
+import { ACTIONS } from '../../constants/fio';
```
Comment (trukhilio) [inline client/src/components/PurchaseNow/middleware.ts:29]: Move to constants import block

File: client/src/util/cart.ts [hunk 5]
```diff
@@ -199,9 +213,14 @@ export const cartItemsToOrderItems = (
       }
 
       return {
-        action: address
-          ? ACTIONS.registerFioAddress
-          : ACTIONS.registerFioDomain,
+        action:
+          type === CART_ITEM_TYPE.DOMAIN_RENEWAL
+            ? ACTIONS.renewFioDomain
```
Comment (trukhilio) [inline client/src/util/cart.ts:218]: Let's move it to function and return what we need. It's hard to read and if we will add more types and actions that would make a mess.

File: server/src/external/fio.mjs [hunk 6]
```diff
@@ -199,6 +199,21 @@ class Fio {
...
+          ...actionParams,
+          fio_domain: options.domain,
+        };
+        break;
+      }
+      case FIO_ACTIONS.addBundledTransactions: {
+        actionParams = {
+          ...actionParams,
+          fio_address: `${options.address}${FIO_ADDRESS_DELIMITER}${options.domain}`,
+          bundle_sets: 1,
```
Comment (trukhilio) [inline server/src/external/fio.mjs:213]: Let's move `1` to constant

File: server/src/models/Order.mjs [hunk 7]
```diff
@@ -534,7 +539,17 @@ export class Order extends Base {
 
       const fioName = address ? `${address}${FIO_ADDRESS_DELIMITER}${domain}` : domain;
       const feeCollected = bcTx.feeCollected || nativeFio;
-      const { hasCustomDomain } = data || {};
+      const { hasCustomDomain, cartItemId } = data || {};
+      const itemType =
+        action === FIO_ACTIONS.renewFioDomain
```
Comment (trukhilio) [inline server/src/models/Order.mjs:544]: Let's also move it to function

**Review by trukhilio — CHANGES_REQUESTED** [review]
Check comments

File: client/src/components/ManagePageContainer/ManagePageContainer.tsx (hunk 3 above)
Comment (vmkarpenko) [reply client/src/components/ManagePageContainer/ManagePageContainer.tsx:18]: but that is constant

Comment (trukhilio) [reply client/src/components/ManagePageContainer/ManagePageContainer.tsx:18]: Oh, sure! Thanks

File: client/src/util/cart.ts [hunk 8]
```diff
@@ -179,17 +190,33 @@ export const deleteCartItem = ({
   }
 };
 
+export const getActionByCartItem = (
+  type: CartItemType,
+  address: string,
+): string => {
+  return type === CART_ITEM_TYPE.DOMAIN_RENEWAL
```
Comment (trukhilio) [inline client/src/util/cart.ts:197]: Let use if condition instead of ternary

File: server/src/models/Order.mjs [hunk 9]
```diff
@@ -469,6 +474,21 @@ export class Order extends Base {
     };
   }
 
+  static getOrderItemType(orderItem) {
+    const { action, address, data } = orderItem;
+    const { hasCustomDomain } = data || {};
+
+    return action === FIO_ACTIONS.renewFioDomain
```
Comment (trukhilio) [inline server/src/models/Order.mjs:481]: Let's also use `if` and `return` value here too

**Pull request merged**